src/mqtt.c

```diff
@@ -207,7 +208,14 @@ static void on_message (
     }
     vl.values_len = ds->ds_num;
 
-    payload = strdup ((void *) msg->payload);
+    payload = calloc (msg->payloadlen+1, sizeof (*payload));
```

Comment:
Please unify the way this size is specified. Either simply use `msg->payloadlen` (preferred, essentially this will mean to revert to `malloc(msg->payloadlen+1)`) or `msg->payloadlen * sizeof (*payload)` (i.e. adapt `memmove()` below to do that multiplication, too).